app/services/foreman/provision/ssh.rb

```diff
@@ -160,24 +156,19 @@ def check_auth_options
       end
 
       def socket_file
-        @socket_file ||= begin
-          file = Tempfile.new("ssh-socket-#{@uuid}")
-          file.close
-          file.path
-        end
+        @socket_file ||= Rails.env.production? ? "/usr/share/foreman/.ssh/ssh-socket-#{@uuid}" : "~/.ssh/ssh-socket-#{@uuid}"
```

Comment:
In production foreman's home directory is `/usr/share/foreman`. That would make both paths expand to the same absolute path, making the condition a bit pointless?